### ISSUE TITLE
Getting data from multiple tables

### ISSUE THREAD

**Issue by tipcircle** [mevdschee/php-crud-api#551]
Forgive the noob question, but I'm not wrapping my head around the queries and how to make this happen. I've tried several combinations, but I can't seem to get it to work. Any help appreciated.

I have a structure like this
![Capture](https://user-images.githubusercontent.com/10174/59479053-8d7b9280-8e29-11e9-95eb-5442cdf7b8ab.PNG)

What I'm trying to do is work out how to query/search by state and city or state and county and return the state name, county name, city name (if queried by state and city) and doctor info associated with the county.

So basically this:
when State = Michigan AND City = Detroit, Return State name, city name, county name, doctor name, doctor number and doctor info associated with Wayne county. (Detroit is in Wayne county, Michigan.)

when State = Michigan AND county = Wayne, Return State name, county name, doctor name, doctor number and doctor info associated with Wayne county.

Hope that makes sense.

Many thanks!

**Comment by mevdschee** [comment]
I think this is what you are looking for:

    api.php/tbl_city?join=tbl_county,tbl_doctor&join=tbl_county,tbl_state&filter=name,eq,Detroit

Kind regards, Maurits

**Comment by tipcircle** [comment]
@mevdschee Thank you very much!

So, here's what the queries return.

By city

/api.php/records/tbl_city?join=tbl_county,tbl_doctor&join=tbl_county,tbl_state&filter=name,eq,Marysville

{
	records: [
		{
			Id: 13,
			county_id: 1,
			name: "Marysville",
			zip_code: "48061"
		}
	]
}

So, it's not retrieving the dr name or county name


And by county

/api.php/records/tbl_county?join=tbl_doctor&join=tbl_state&filter=name,eq,Saint%20Clair

{
	records: [
		{
			Id: 1,
			state_id: 1,
			name: "Saint Clair",
			population: 160469,
			doctor_id: 1
		}
	]
}

So, it's not including the dr name

But, that is definitely a lot closer to the data I'm trying to get at :-) Data structuring is definitely not my forte.

Is there a way to filter on the included tables or another way to get at the data?

> NB: What is column state_id doing in tbl_state? And is there really only one doctor per county?

In this instance, yes, there will only be one doctor per county. Luckily that's not really the case irl :-)

It's used as the primary key. Maybe these images will be of help.

![Pasted File at June 11, 2019 11_47 AM](https://user-images.githubusercontent.com/10174/59506580-92b2fe80-8e76-11e9-8e59-20b3e8de2475.png)

![Pasted File at June 11, 2019 11_48 AM](https://user-images.githubusercontent.com/10174/59506590-96df1c00-8e76-11e9-8cf6-c8ad3df682e4.png)

Thank you, much appreciated

**Comment by tipcircle** [comment]
I decided to go another route with this but still using your awesome software. My database is just designed incorrectly for what I'd like to do. Thanks!

**Comment by mevdschee** [comment]
>So, it's not retrieving the dr name or county name

You should add a foreign key constraint on "county_id" and "doctor_id" in order to get the nested results.

See: https://www.w3schools.com/sql/sql_foreignkey.asp

**Comment by tipcircle** [comment]
Thank you!

**Comment by tipcircle** [comment]
Thanks for your help, much appreciated. I am able to return the results I need with these two queries, using the original database structure.

By County:

`select tbl_state.name, tbl_county.name, tbl_county.doctor_id, tbl_doctor.name, tbl_doctor.phone, tbl_doctor.info 
  from ((tbl_state inner join tbl_county on tbl_state.Id = tbl_county.state_id)
                  inner join tbl_doctor on tbl_doctor.Id = tbl_county.doctor_id)
                  where tbl_state.name = 'Michigan' and tbl_county.name = 'Saint Clair';`


And by city:

`select tbl_state.name, tbl_county.name, tbl_county.doctor_id, tbl_doctor.name, tbl_doctor.phone, tbl_doctor.info 
  from (((tbl_state inner join tbl_county on tbl_state.Id = tbl_county.state_id)
                  inner join tbl_city on tbl_county.Id = tbl_city.county_id)
                  inner join tbl_doctor on tbl_doctor.Id = tbl_county.doctor_id)
                  where tbl_state.name = 'Michigan' and tbl_city.name = 'Yale';`

Are these queries possible using your software?

Thank you

**Comment by mevdschee** [comment]
No, these are not exactly possible, as you can not have a where clause on joined tables. I do doubt that you need to require in your where that 'Saint Clair' is in 'Michigan' or that 'Yale' is in 'Michigan'. Is it because your city/county names are not unique? Anyway.. even in that case you can solve it with multiple API requests (which is not very inefficient).

Note that I already gave you the queries, but here you go again:

by city:

    api.php/tbl_city?join=tbl_county,tbl_doctor&join=tbl_county,tbl_state&filter=name,eq,Yale

or by county:

    api.php/tbl_county?join=tbl_doctor&join=tbl_state&filter=name,eq,Saint Clair

I hope this helps. Kind regards,  Maurits

**Comment by tipcircle** [comment]
> Did you add the foreign key constraints? Note that they are required for the join to work
Not yet, I'm gong to try that next.

> Is it because your city/county names are not unique?
Correct

Anyway, thanks for your help, it's much appreciated. I'm determined to get this to work because I really like this program you wrote and I'd really like to use it.

Any place to donate a few bucks to the further development?

**Comment by mevdschee** [comment]
Closed due to inactivity.
